### PR TITLE
#6176: add timeout to non-interactive legacy google token retrieval

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Part of #6176 
- Adds a timeout to non-interactive Google token retrieval
- Probably doesn't solve the underlying cause of the hanging

## Checklist

- [ ] Add tests
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

**Review by grahamlangford — APPROVED** [review]

**Review by BLoe — APPROVED** [review]

**Review by grahamlangford — APPROVED** [review]

File: src/contrib/google/sheets/core/sheetsApi.ts [hunk 1]
```diff
@@ -80,20 +81,41 @@ async function _ensureSheetsReadyOnce({
...
+ * Ensure the Google Sheets API is ready to be called, and return the user's token.
+ * @param maxRetries the maximum number of retries
+ * @param interactive true to show the Google authentication UI if needed
+ * @param timeoutMillis a timeout for ensuring the token if the Google API is not ready
+ * @return token the user's Google token
+ */
 export async function ensureSheetsReady({
   maxRetries = 3,
   interactive,
+  timeoutMillis = 1000,
```
Comment (BLoe) [inline src/contrib/google/sheets/core/sheetsApi.ts:94]: @twschiller Any concerns here about 1 second as too short a timeout for the init script to run on a slow machine/network?

Comment (twschiller) [reply src/contrib/google/sheets/core/sheetsApi.ts:94]: I don't think there should be a network call involved - this is just getting the token from the Chrome API

Comment (BLoe) [reply src/contrib/google/sheets/core/sheetsApi.ts:94]: It also may call `initGoogle()` if it hasn't been called yet:

https://github.com/pixiebrix/pixiebrix-extension/blob/main/src/contrib/google/sheets/core/sheetsApi.ts#L68-L74

Comment (BLoe) [reply src/contrib/google/sheets/core/sheetsApi.ts:94]: Which, I'm almost positive does involve a network request to dynamically load the gsheets client, right?

**Pull request merged**